src/test/java/com/commercetools/sync/products/helpers/PriceReferenceResolverTest.java

```diff
@@ -199,14 +208,90 @@ public void resolveChannelReference_WithChannelKeyAsUuidSetAndAllowed_ShouldReso
+            new PriceReferenceResolver(productSyncOptions, typeService, channelService, customerGroupService);
+
+        final PriceDraftBuilder resolvedBuilder = priceReferenceResolver.resolveChannelReference(priceBuilder)
+                                                                        .toCompletableFuture().join();
+        assertThat(resolvedBuilder.getChannel()).isNotNull();
+        assertThat(resolvedBuilder.getChannel().getId()).isEqualTo(CHANNEL_ID);
+    }
+
+    @Test
+    public void resolveChannelReference_WithNonExistingChannelKey_ShouldResolveChannelReference() {
+        final ProductSyncOptions productSyncOptions = ProductSyncOptionsBuilder.of(mock(SphereClient.class))
+                                                                               .allowUuidKeys(true)
+                                                                               .build();
+        final PriceDraftBuilder priceBuilder = PriceDraftBuilder
+            .of(MoneyImpl.of(BigDecimal.TEN, DefaultCurrencyUnits.EUR))
+            .country(CountryCode.DE)
+            .channel(Channel.referenceOfId("channelKey"));
+
+        final PriceReferenceResolver priceReferenceResolver =
+            new PriceReferenceResolver(productSyncOptions, typeService, channelService, customerGroupService);
 
         final PriceDraftBuilder resolvedBuilder = priceReferenceResolver.resolveChannelReference(priceBuilder)
-                                                               .toCompletableFuture().join();
+                                                                        .toCompletableFuture().join();
         assertThat(resolvedBuilder.getChannel()).isNotNull();
         assertThat(resolvedBuilder.getChannel().getId()).isEqualTo(CHANNEL_ID);
     }
 
+    @Test
+    public void
+        resolveSupplyChannelReference_WithNonExistingChannelAndNotEnsureChannel_ShouldNotResolveChannelReference() {
+        when(channelService.fetchCachedChannelId(anyString()))
+            .thenReturn(CompletableFuture.completedFuture(Optional.empty()));
+
+        final PriceDraftBuilder priceBuilder = PriceDraftBuilder
+            .of(MoneyImpl.of(BigDecimal.TEN, DefaultCurrencyUnits.EUR))
+            .country(CountryCode.DE)
+            .channel(Channel.referenceOfId("channelKey"));
+
+        final PriceReferenceResolver priceReferenceResolver =
+            new PriceReferenceResolver(syncOptions, typeService, channelService, customerGroupService);
+
+
+        priceReferenceResolver.resolveChannelReference(priceBuilder)
+                              .exceptionally(exception -> {
+                                  assertThat(exception).isExactlyInstanceOf(CompletionException.class);
+                                  assertThat(exception.getCause())
+                                      .isExactlyInstanceOf(ReferenceResolutionException.class);
+                                  assertThat(exception.getCause().getCause())
```

Comment:
Why is it wrapped twice?

Reply:
@lojzatran because there are two causes for the exception:
1. Reference resolution exception "Failed to resolve the `channel` reference on the PriceDraft..."
2. Channel with key 'channelKey' doesn't exits.